populate/src/main/java/bio/terra/pearl/populate/service/extract/StudyExtractor.java

```diff
@@ -0,0 +1,139 @@
+        }
+    }
+
+    public StudyEnvironmentPopDto extractStudyEnv(StudyEnvironment studyEnv,
+                                                  StudyPopDto studyPopDto,
+                                                  ExtractPopulateContext context) {
+        StudyEnvironmentPopDto studyEnvPopDto = new StudyEnvironmentPopDto();
+        studyEnvPopDto.setEnvironmentName(studyEnv.getEnvironmentName());
+        studyEnvPopDto.setStudyEnvironmentConfig(
+                studyEnvironmentConfigService.find(studyEnv.getStudyEnvironmentConfigId()).orElseThrow(StudyEnvConfigMissing::new)
+        );
+        if (studyEnv.getPreEnrollSurveyId() != null) {
+            SurveyExtractor.SurveyPopDtoStub surveyPopDtoStub = new SurveyExtractor.SurveyPopDtoStub();
+            surveyPopDtoStub.setPopulateFileName("../../" + context.getFileNameForEntity(studyEnv.getPreEnrollSurveyId()));
+            studyEnvPopDto.setPreEnrollSurveyDto(surveyPopDtoStub);
+        }
+        List<StudyEnvironmentSurvey> studyEnvSurveys = studyEnvironmentSurveyService.findAllByStudyEnvId(studyEnv.getId(), null);
+        for (StudyEnvironmentSurvey studyEnvSurvey : studyEnvSurveys) {;
+            StudyEnvironmentSurveyPopDto studyEnvSurveyPopDto = new StudyEnvironmentSurveyPopDto();
+            BeanUtils.copyProperties(studyEnvSurvey, studyEnvSurveyPopDto, "id", "studyEnvironmentId", "surveyId");
+            studyEnvSurveyPopDto.setPopulateFileName("../../" + context.getFileNameForEntity(studyEnvSurvey.getSurveyId()));
+            studyEnvPopDto.getConfiguredSurveyDtos().add(studyEnvSurveyPopDto);
+        }
+        List<StudyEnvironmentConsent> studyEnvConsents = studyEnvironmentConsentService.findAllByStudyEnvironmentId(studyEnv.getId());
+        for (StudyEnvironmentConsent studyEnvConsent : studyEnvConsents) {;
+            StudyEnvironmentConsentPopDto consentPopDto = new StudyEnvironmentConsentPopDto();
+            BeanUtils.copyProperties(studyEnvConsent, consentPopDto, "id", "studyEnvironmentId", "consentFormId");
+            String filename = "../../" + context.getFileNameForEntity(studyEnvConsent.getConsentFormId());
+            consentPopDto.setPopulateFileName(filename);
+            studyEnvPopDto.getConfiguredConsentDtos().add(consentPopDto);
+        }
+
+        List<NotificationConfig> notificationConfigs = notificationConfigService.findByStudyEnvironmentId(studyEnv.getId());
+        for (NotificationConfig config : notificationConfigs) {;
+            NotificationConfigPopDto configPopDto = new NotificationConfigPopDto();
+            BeanUtils.copyProperties(config, configPopDto, "id", "studyEnvironmentId", "portalEnvironmentId", "emailTemplateId");
+            String filename = "../../" + context.getFileNameForEntity(config.getEmailTemplateId());
+            configPopDto.setPopulateFileName(filename);
+            studyEnvPopDto.getNotificationConfigDtos().add(configPopDto);
+        }
+
+        // TODO extract dashboard configs
```

Comment:
I'm curious, why were dashboard configs and kit configuration types left out of this PR?

Reply:
just didn't have time, and wanted to get this in so I could update the demo and development stuff with prod HeartHive configs.  I'll make a ticket for it